File: frontend/web/src/components/conversation/persona.test.ts
import { describe, expect, it } from 'vitest'

import { personaForAgent, personaForAuthor } from './persona'

describe('personaForAgent', () => {
  it('returns Aveline for the aveline key', () => {
    expect(personaForAgent('aveline').name).toBe('Aveline')
  })

  it('returns Ava for the ava key', () => {
    expect(personaForAgent('ava').name).toBe('Ava')
  })

  it('returns Elle for the elle key', () => {
    expect(personaForAgent('elle').name).toBe('Elle')
  })

  it('returns Lina for the lina key', () => {
    expect(personaForAgent('lina').name).toBe('Lina')
  })

  /**
   * Lina's accent must not go through `commerce`.
   *
   * `commerce` is the brand wine-rose and is also what paints the documentation section, the
   * pricing CTA and the orders/payments surfaces. Routing her accent through it meant an
   * agent-colour change would recolour all of those, and it also made her avatar identical to
   * Aveline's, who uses `primary` - the same value. She has her own lilac token instead.
   */
  describe("Lina's accent", () => {
    it('uses the lilac token rather than commerce or primary', () => {
      const lina = personaForAgent('lina')
      expect(lina.bg).toBe('bg-lilac')
      expect(lina.text).toBe('text-lilac')
      expect(lina.bgSoft).toBe('bg-lilac/10')
      expect(lina.ring).toBe('ring-lilac/20')
    })

    it('is distinct from every other persona accent', () => {
      const accents = ['aveline', 'ava', 'elle', 'lina'].map((key) => personaForAgent(key).bg)
      expect(new Set(accents).size).toBe(4)
    })

    it('leaves Aveline on the brand primary and the others on their own tokens', () => {
      expect(personaForAgent('aveline').bg).toBe('bg-primary')
      expect(personaForAgent('ava').bg).toBe('bg-memory')
      expect(personaForAgent('elle').bg).toBe('bg-visual')
    })
  })

  it('defaults to Aveline for an unknown key', () => {
    expect(personaForAgent('unknown').name).toBe('Aveline')
  })

  it('defaults to Aveline for null/undefined', () => {
    expect(personaForAgent(null).name).toBe('Aveline')
    expect(personaForAgent(undefined).name).toBe('Aveline')
  })
})

describe('personaForAuthor', () => {
  it('returns the persona for an Agent author', () => {
    expect(personaForAuthor('Agent', 'elle')?.name).toBe('Elle')
  })

  it('returns null for a User author', () => {
    expect(personaForAuthor('User', null)).toBeNull()
  })

  it('returns null for a System author', () => {
    expect(personaForAuthor('System', null)).toBeNull()
  })
})

File: frontend/aveline_mobile/test/shared/persona_test.dart
import 'package:aveline_mobile/shared/persona.dart';
import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

/// Lina's accent is her own colour, not the theme seed.
///
/// She used to share Aveline's wine-rose (`0xFF8B2E42`), which is also `AppTheme.seedColor` and
/// the colour the docs chrome and the pricing CTA are built on, so her avatar rendered as the
/// app's own accent rather than hers. This pins the separation in both directions: her accent is
/// the lilac, and nobody else's moved.
void main() {
  group('personaForAgent', () {
    test('gives Lina her own lilac', () {
      final lina = personaForAgent('lina');
      expect(lina.name, 'Lina');
      expect(lina.role, 'Commerce');
      expect(lina.accent, const Color(0xFF6B5A9E));
    });

    test('no longer gives Lina the theme seed', () {
      expect(personaForAgent('lina').accent, isNot(const Color(0xFF8B2E42)));
    });

    test('leaves every other agent accent untouched', () {
      expect(personaForAgent('aveline').accent, const Color(0xFF8B2E42));
      expect(personaForAgent('ava').accent, const Color(0xFFB0566B));
      expect(personaForAgent('elle').accent, const Color(0xFFB08D57));
    });

    test('gives all four agents distinct accents', () {
      final accents = {
        for (final key in ['aveline', 'ava', 'elle', 'lina'])
          personaForAgent(key).accent,
      };
      expect(accents.length, 4);
    });

    test('defaults to Aveline for an unknown or null key', () {
      expect(personaForAgent('nobody').name, 'Aveline');
      expect(personaForAgent(null).name, 'Aveline');
    });
  });

  group('personaForAuthor', () {
    test('returns the persona for an Agent author', () {
      expect(personaForAuthor('Agent', 'lina')?.name, 'Lina');
    });

    test('returns null for a non-agent author', () {
      expect(personaForAuthor('User', 'lina'), isNull);
      expect(personaForAuthor('System', null), isNull);
    });
  });
}
